### PR TITLE
Dataview sync issue

### PR DESCRIPTION
**Pull request by milesAraya**
This is replica of https://github.com/arayabrain/optinist-for-cloud/pull/302 rebased onto [combine-alerts-branches](https://github.com/arayabrain/optinist-for-cloud/tree/combine-alerts-branches)

## Issue being fixed
Startup sync gap - When a new ECS task spins up, API worker containers without the scheduler never download published experiments. The data was only downloaded once a user logs in and tries to access the data. However, the public site was accessble before any user logged in. Requests hitting those containers would serve stale/missing data until the scheduled sync container happened to run. 

The idea is to download on instance startup automatically, in the background. 

## Summary of changes

- Add one-time startup sync so API worker containers download missing published experiments on boot (independent of the background scheduler)
- Defensively handle empty EC2 `Reservations` responses in cleanup job to avoid `IndexError` on deregistered instances
- Migrate raw SQL queries in `StorageReconciliationJob` to SQLModel/SQLAlchemy ORM
- Extract magic-number concurrency limits into class-level constants on `PublishedExperimentSyncJob`

---

## Changes by File

**`studio/__main_unit__.py`**
- Schedule `run_startup_sync()` as a background `asyncio.Task` during lifespan startup (5s delay, stored on `app.state` to prevent GC)

**`studio/app/common/core/background/sync_job.py`**
- Add `run_startup_sync()`: queries ALL published experiments, checks local file existence, downloads missing ones in two phases (thumbnails then metadata) without modifying DB sync status
- Add `_get_all_published_experiments()`: queries published experiments ignoring `local_sync_status`
- Extract `THUMBNAIL_CONCURRENCY = 10` and `METADATA_CONCURRENCY = 3` as class constants, replace inline magic numbers in `_sync_thumbnails` and `_sync_metadata`

**`studio/app/common/core/background/cleanup_job.py`**
- Guard `describe_instances` response with `.get()` checks for empty `Reservations` / `Instances`; treat missing reservation data as terminated and clean up the orphaned assignment

**`studio/app/common/core/background/storage_reconciliation_job.py`**
- Replace three raw SQL strings with equivalent `select()` / `func.count()` / `or_()` ORM expressions using `UserStorageUsage` model

**`studio/tests/app/common/core/background/test_cleanup_job.py`**
- Add `test_handle_orphaned_data_empty_reservations` covering the new empty-reservations guard

**`studio/tests/app/common/core/background/test_sync_job.py`**
- Add `TestStartupSync` class with three tests: downloads missing experiments, skips locally-present ones, handles empty published list

### CODE REVIEW

File: studio/app/common/core/background/sync_job.py [hunk 1]
```diff
@@ -80,6 +83,99 @@ async def run(cls):
         except Exception as e:
             logger.error(f"Fatal error in sync job: {e}", exc_info=True)
 
+    @classmethod
+    async def run_startup_sync(cls):
```
Comment (itutu-tienday) [inline studio/app/common/core/background/sync_job.py:87]: This is my opinion on code refactoring.

I understand that this process operates differently from the normal data synchronization process (PublishedExperimentSyncJob.run), but I think there are some common aspects to the process flow.

The current code structure seems to be somewhat different between run and run_startup_sync, but standardizing the common parts would likely improve readability and maintainability.

- Concerns
  - run is internally composed of two steps, _sync_thumbnails and _run_sync_logic, and is concise.
  - Wouldn't run_startup_sync have a similar structure to the above?

Please consider this.

**Review by itutu-tienday — APPROVED** [review]
There are suggestions for code refactoring (priority: low), but I will approve it so that deployment and testing can take priority.
